### PR TITLE
feat(vertical-swipe-view): UX & logic

### PR DESCRIPTION
**Pull request by VladislavRiabko**
- add VerticalSwipeView component 
- add scroll down logic 
- add swipes logic

### CODE REVIEW

File: app/components/VerticalSwipeView/VerticalSwipeView.props.ts [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { ImageSourcePropType } from 'react-native';
+
+export interface IVerticalSwipeViewProps {
+  renderTopElement: () => JSX.Element;
+  renderBottomElement: () => JSX.Element;
+  backgroundImage: ImageSourcePropType;
```
Comment (emberglaz) [inline app/components/VerticalSwipeView/VerticalSwipeView.props.ts]: use 
```
title: 
subtitle: 
image: 
etc...
```

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          scrollEnabled={false}
+          ref={scrollViewRef}
+          style={styles.scrollView}>
+          <View style={styles.topContentContainer}>
+            {renderTopElement && renderTopElement()}
+            <ExtendedButton title={t('some')} onPress={onDownPress} />
+          </View>
+          <View style={styles.bottomContentContainer}>
+            {renderBottomElement && renderBottomElement()}
+            <ExtendedButton title={t('some')} onPress={onUpPress} />
```
Comment (emberglaz) [inline app/components/VerticalSwipeView/VerticalSwipeView.tsx]: use correct localization key

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  return (
+    <View style={generalStyles.flex}>
+      <ImageBackground source={backgroundImage} style={generalStyles.flex}>
+        <ScrollView
+          scrollEnabled={false}
+          ref={scrollViewRef}
+          style={styles.scrollView}>
+          <View style={styles.topContentContainer}>
+            {renderTopElement && renderTopElement()}
+            <ExtendedButton title={t('some')} onPress={onDownPress} />
```
Comment (emberglaz) [inline app/components/VerticalSwipeView/VerticalSwipeView.tsx]: use correct localization key

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+  const scrollViewRef = useRef<ScrollView>(null);
+
+  const onDownPress = useCallback(() => {
+    scrollViewRef.current?.scrollToEnd();
+  }, []);
+
+  const onUpPress = useCallback(() => {
+    scrollViewRef.current?.scrollTo({ x: 0, y: 0, animated: true });
+  }, []);
```
Comment (emberglaz) [inline app/components/VerticalSwipeView/VerticalSwipeView.tsx]: create a state for the scroll position and create one function to change the scroll through a boolean operator

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+
+  return (
+    <View style={generalStyles.flex}>
+      <ImageBackground source={backgroundImage} style={generalStyles.flex}>
+        <ScrollView
+          scrollEnabled={false}
+          ref={scrollViewRef}
+          style={styles.scrollView}>
+          <View style={styles.topContentContainer}>
+            {renderTopElement && renderTopElement()}
```
Comment (emberglaz) [inline app/components/VerticalSwipeView/VerticalSwipeView.tsx]: in this component you will always have top and bottom content
rework the logic

**Review by emberglaz — CHANGES_REQUESTED** [review]

File: app/components/VerticalSwipeView/VerticalSwipeView.styles.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    height: SCROLL_VIEW_HEIGHT,
+  },
+  topContentContainer: {
+    height: SCROLL_VIEW_HEIGHT / 2,
+    justifyContent: 'flex-end',
+  },
+  bottomContentContainer: {
+    height: SCROLL_VIEW_HEIGHT / 2,
+  },
+});
```
Comment (Reykjavik151) [inline app/components/VerticalSwipeView/VerticalSwipeView.styles.ts]: Test it with small / big test devices (Android / iOS).

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx (hunk 5 above)
Comment (Reykjavik151) [inline app/components/VerticalSwipeView/VerticalSwipeView.tsx]: We should support swipe logic, please take a look for it with `react-native-gesture-handler` GestureDetector:
https://docs.swmansion.com/react-native-gesture-handler/docs/gesture-composition

File: app/components/VerticalSwipeView/VerticalSwipeView.data.ts [hunk 7]
```diff
@@ -0,0 +1,3 @@
+import { WINDOW_HEIGHT } from '../../constants/window';
+
+export const SCROLL_VIEW_HEIGHT = (WINDOW_HEIGHT - 78) * 2;
```
Comment (Reykjavik151) [inline app/components/VerticalSwipeView/VerticalSwipeView.data.ts]: What is the 78 magic number? :D 
Did you test it with Android?
Please, make this number as constant here (without the export maybe)

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx (hunk 5 above)
Comment (emberglaz) [reply app/components/VerticalSwipeView/VerticalSwipeView.tsx]: not now

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+  const scrollViewRef = useRef<ScrollView>(null);
+
+  const setScrollPosition = useCallback(() => {
+    if (isTopPosition) {
+      scrollViewRef.current?.scrollToEnd();
+      setIsTopPosition(false);
+    } else {
+      scrollViewRef.current?.scrollTo({ x: 0, y: 0, animated: true });
+      setIsTopPosition(true);
+    }
```
Comment (Reykjavik151) [inline app/components/VerticalSwipeView/VerticalSwipeView.tsx]:
```
if (isTopPosition) {
   scrollViewRef.current?.scrollToEnd();
 } else {
   scrollViewRef.current?.scrollTo({ x: 0, y: 0, animated: true });
 }
  setIsTopPosition(!isTopPosition);
     ```

**Review by Reykjavik151 — CHANGES_REQUESTED** [review]

File: app/components/VerticalSwipeView/VerticalSwipeView.tsx (hunk 8 above)
Comment (Reykjavik151) [reply app/components/VerticalSwipeView/VerticalSwipeView.tsx]: fix it and you can merge by yourself

**Pull request merged**